libsolidity/ast/Types.h

```diff
@@ -50,6 +51,8 @@ using TypePointer = std::shared_ptr<Type const>;
 using FunctionTypePointer = std::shared_ptr<FunctionType const>;
 using TypePointers = std::vector<TypePointer>;
 using rational = boost::rational<dev::bigint>;
+using TypeResult = Result<TypePointer>;
+using BoolResult = Result<bool>;
```

Comment:
Looks very Rust-y.

Reply:
I am not too sure yet whether `Result<bool>` (and friends) is absolutely sufficient to write and does not require a type alias. But I may be opinionated here. :-)

Reply:
I think we can leave it like that because it would work fine for ``Result<bool>``, but would need more typing for Result<TypePointer>.